src/esp32/esp32_bt_gattc.c

```diff
@@ -72,21 +84,23 @@ bool mgos_bt_gattc_subscribe(int conn_id, uint16_t handle) {
   if (esp32_bt_is_scanning()) return false;
   struct conn *conn = find_by_conn_id(conn_id);
   if (conn == NULL) return false;
+  last_subscribe_conn_id = conn_id;
   esp_err_t err =
       esp_ble_gattc_register_for_notify(conn->iface, conn->c.addr.addr, handle);
-  /*
-   * This is not enough, must write to the corresponding descriptor
-   * to actually enable notifications on the remote side.
-   * TODO(lsm): Implement
-   */
   return err == ESP_OK;
 }
 
 bool mgos_bt_gattc_write(int conn_id, uint16_t handle, const void *data,
                          int len) {
   if (esp32_bt_is_scanning()) return false;
-  /* TODO(lsm): implement */
-  return false;
+  struct conn *conn = find_by_conn_id(conn_id);
+  if (conn == NULL) return false;
+  uint8_t *dd = calloc(len, 1);
```

Comment:
is this copy necessary? dd is freed immediately after the call, so i don't really see the point. can you elaborate on why you're doing it?

Reply:
It is there because `esp_ble_gattc_write_char` expect pointer to non-const data. Otherwise I get:

```
/home/martin/app/deps/bt-common/src/esp32/esp32_bt_gattc.c:101:44: error: passing argument 5 of 'esp_ble_gattc_write_char' discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
                                            data, ESP_GATT_WRITE_TYPE_RSP, 0);
```

Reply:
i see. yes, their api could use a sprinkling of const here and there. the argument is not modified though, so you should eb able to just cast the const away and it'll work.